src/Skia/Avalonia.Skia/Text/SKTextLayout.cs

```diff
@@ -0,0 +1,1602 @@
+                    }
+                    else
+                    {
+                        length += lineBreakPosition - currentPosition + 1;
+
+                        var splitResult = SplitTextRuns(text, currentTextRuns, length);
+
+                        var textLine = CreateShapedTextLine(text, splitResult.FirstTextRuns, currentPosition);
+
+                        var textWrappingResult = PerformTextWrapping(text, textLine);
+
+                        textLines.AddRange(textWrappingResult);
+
+                        currentTextRuns = splitResult.SecondTextRuns;
+
+                        currentPosition += textLine.TextPointer.Length;
+
+                        break;
+                    }
+                }
+            }
+
+            if (spans != null)
+            {
+                foreach (var textStyleSpan in spans)
+                {
+                    ApplyTextStyleSpan(text, textLines, textStyleSpan);
+                }
+            }
+
+            var sizeX = 0.0f;
+            var sizeY = 0.0f;
+
+            foreach (var textLine in textLines)
+            {
+                if (sizeX < textLine.LineMetrics.Size.Width)
+                {
+                    sizeX = textLine.LineMetrics.Size.Width;
+                }
+
+                sizeY += textLine.LineMetrics.Size.Height;
+            }
+
+            Size = new Size(sizeX, sizeY);
+
+            return textLines;
+        }
+
+        /// <summary>
+        /// Creates a text run with a specific text format.
+        /// </summary>
+        /// <param name="text">The text.</param>
+        /// <param name="textPointer">The text pointer.</param>
+        /// <param name="textFormat">The text format.</param>
+        /// <param name="foreground">The foreground brush.</param>
+        /// <param name="shapeRun">Indicates whether the run should get shaped or not.</param>
+        /// <returns></returns>
+        private SKTextRun CreateTextRun(ReadOnlySpan<char> text, SKTextPointer textPointer, SKTextFormat textFormat, IBrush foreground = null, bool shapeRun = false)
+        {
+            _paint.Typeface = textFormat.Typeface;
+
+            _paint.TextSize = textFormat.FontSize;
+
+            _paint.TextEncoding = SKTextEncoding.Utf16;
+
+            var fontMetrics = _paint.FontMetrics;
+
+            if (textPointer.Length == 0)
+            {
+                return CreateEmptyTextRun(textFormat);
+            }
+
+            if (!shapeRun)
+            {
+                return new SKTextRun(textPointer, null, textFormat, fontMetrics, 0, foreground);
+            }
+
+            using (var shaper = new SKShaper(textFormat.Typeface))
+            using (var buffer = new HarfBuzzSharp.Buffer())
+            {
+                buffer.AddUtf16(text, (uint)textPointer.StartingIndex, textPointer.Length);
+
+                buffer.GuessSegmentProperties();
+
+                var result = shaper.Shape(buffer, _paint);
+
```

Comment:
Don't use the result and instead use the buffer directly to get all glyph advances etc